integration-tests/src/test/java/org/jboss/pnc/bacon/test/AbstractTest.java

```diff
@@ -64,28 +65,33 @@ protected static String getRandomString() {
     }
 
     protected void execute(String... args) {
-        ExecutionResult result = executor.runCommand(args);
+        ExecutionResult result = executor.runCommand(null, args);
         assertThat(result.getOutput()).isEmpty();
         assertThat(result.getError()).isEmpty();
         assertThat(result.getRetval()).isZero();
     }
 
     protected <T> T executeAndDeserialize(Class<T> clazz, String... args) throws JsonProcessingException {
-        ExecutionResult result = executor.runCommand(args);
+        ExecutionResult result = executor.runCommand(null, args);
         log.debug("stderr:{}{}", System.lineSeparator(), result.getError());
         assertThat(result.getRetval()).isZero();
         return result.fromYAML(clazz);
     }
 
-    protected <T> T executeAndDeserializeJSON(Class<T> clazz, String... args) throws JsonProcessingException {
-        ExecutionResult result = executor.runCommand(args);
+    protected <T> T executeAndDeserializeJSON(Class<T> clazz, Path workingDir, String... args)
+            throws JsonProcessingException {
+        ExecutionResult result = executor.runCommand(workingDir, args);
         log.debug("stderr:{}{}", System.lineSeparator(), result.getError());
         assertThat(result.getRetval()).isZero();
         return result.fromYAML(clazz);
     }
 
     protected ExecutionResult executeAndGetResult(String... args) {
-        return executor.runCommand(args);
+        return executor.runCommand(null, args);
```

Comment:
So this sets the `dir` argument of `Runtime.exec()` to `null` which means `$CWD`, but I still don't really like that it will create this dir anywhere I run it. I think picking a single location would be better.

The other issue is because it's not a unique location, it would be possible to run the `run` with one project and `release` with another project using the same `.bacon/` directory, and what would happen then?

I kind of like `$HOME/.bacon/<project_id>/` or something similar.

Reply:
Great points - I think we need to understand its purpose before choosing a solution.

Reply:
By default the .bacon directory it's created where the program is executed. The location can be overriden with an environment varialbe and a system property.
Current directory was chosen as a default to allow running the tool for multiple projects separately.

If the config yaml or any other file in the config dir would change, the context will be dropped (the previous one won't be read and will be overwritten at the end of the execution). It's checked by storing a hash sum of all files in the directory (and their locations, moving a file is also considered a change).

`executeAndGetResult` is also used for non-pig tests, for it I haven't changed the working directory (hence the null here)

Reply:
I think it would be better to use a Junit tmp dir here and pass that through?

Reply:
Ah, I've just seen down below you are passing a TempDir in :-) What about changing for all tests so they consistently run from a tempDirectory - or do some require a certain context pwd?